### PR TITLE
Add missing unit tests

### PR DESCRIPTION
**Pull request by jonathanmos**
### What does this PR do?

Add missing unit tests for TextValueResolver, and also for StringObfuscator implementations

### CODE REVIEW

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/FixedLengthStringObfuscatorTest.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @BeforeEach
+    fun `set up`() {
+        testedObfuscator = FixedLengthStringObfuscator()
+    }
+
+    @Test
+    fun `M return STATIC_MASK W obfuscate(){string}`(forge: Forge) {
+        // Given
+        val fakeInputString: String = forge.aString()
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/FixedLengthStringObfuscatorTest.kt:38]: nit: no need for the explicit type of `fakeInputString` (and in other tests).

another way to generate `fakeInputString` is to use argument injections:

```
fun `M return STATIC_MASK W obfuscate(){string}`(@StringForgery fakeInputString: String) {
```

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/TextValueResolverTest.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        val mockTextView: TextView = forge.aMockTextView().apply {
+            whenever(this.text).thenReturn("")
+            whenever(this.hint).thenReturn("")
+        }
+
+        // When
+        val resolvedText: String = testedResolver.resolveTextValue(mockTextView)
+
+        // Then
+        assertThat(resolvedText).isEqualTo("")
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/TextValueResolverTest.kt:52]: alternatively can be just `isEmpty`

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/TextValueResolverTest.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        val fakeExpectedText: String = forge.aString()
+        val fakeExpectedHint: String = forge.aString()
+
+        val mockTextView: TextView = forge.aMockTextView().apply {
+            whenever(this.text).thenReturn(fakeExpectedText)
+            whenever(this.hint).thenReturn(fakeExpectedHint)
+        }
+
+        // When
+        val resolvedText: String = testedResolver.resolveTextValue(mockTextView)
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/TextValueResolverTest.kt:101]: is there any reason for explicit type?

**Review by 0xnm — APPROVED** [review]
congratulations with first contribution! I left some comments, but they are opinionated, so everything lgtm.

**Review by xgouchet — APPROVED** [review]

**Pull request merged**